### PR TITLE
fix help menu for makefile

### PR DESCRIPTION
**Pull request by weshayutin**
* help description was located in the wrong place or missing in some commands
* clean up outdated developer docs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -417,14 +406,14 @@ sed -r "s/[&]* [!] $(CLUSTER_TYPE)|[!] $(CLUSTER_TYPE) [&]*//")) || $(CLUSTER_TY
 #TEST_FILTER := $(shell echo '! aws && ! gcp && ! azure' | sed -r "s/[&]* [!] $(CLUSTER_TYPE)|[!] $(CLUSTER_TYPE) [&]*//")
 SETTINGS_TMP=/tmp/test-settings
 
-test-e2e-setup:
+test-e2e-setup:
```
Comment (kaovilai) [inline Makefile:409]: No space needed here :)

File: docs/developer/local_dev.md [hunk 2]
```diff
@@ -24,5 +24,5 @@
 3. Deploy OADP operator using your image:
 
     ```
-    IMG=quay.io/<CONTAINER_REGISTRY_USERNAME>/oadp-operator:<IMAGE_TAG> make deploy
+    IMG=quay.io/<CONTAINER_REGISTRY_USERNAME>/oadp-operator:<IMAGE_TAG> make docker-push
```
Comment (kaovilai) [inline docs/developer/local_dev.md:27]: I would actually delete this file in favor of install_from_source.md

docker-push here won't be doing much help with creating an installable bundle.

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: docs/developer/local_dev.md (hunk 2 above)
Comment (weshayutin) [reply docs/developer/local_dev.md:27]: so should I just nuke developer/local_dev.md ?

**Comment by kaovilai** [comment]
On Fri, Sep 9, 2022, 11:59 AM Wesley Hayutin ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In docs/developer/local_dev.md
> <https://github.com/openshift/oadp-operator/pull/817#discussion_r967234197>
> :
>
> > @@ -24,5 +24,5 @@
>  3. Deploy OADP operator using your image:
>
>      ```
> -    IMG=quay.io/<CONTAINER_REGISTRY_USERNAME>/oadp-operator:<IMAGE_TAG> make deploy
> +    IMG=quay.io/<CONTAINER_REGISTRY_USERNAME>/oadp-operator:<IMAGE_TAG> make docker-push
>
> so should I just nuke developer/local_dev.md ?
>
That would be preferable in my book.

File: docs/developer/local_dev.md (hunk 2 above)
Comment (kaovilai) [reply docs/developer/local_dev.md:27]: That would be preferable for me.

File: Makefile (hunk 1 above)
Comment (weshayutin) [reply Makefile:409]: ah dang.. thanks

File: docs/developer/local_dev.md (hunk 2 above)
Comment (weshayutin) [reply docs/developer/local_dev.md:27]: done. thanks!

**Review by kaovilai — APPROVED** [review]

**Pull request merged**